cli/azd/internal/appdetect/java.go

```diff
@@ -64,6 +64,7 @@ func (jd *javaDetector) DetectProject(ctx context.Context, path string, entries
 				if inRoot := strings.HasPrefix(pomFile, parentPomItem.path); inRoot {
 					parentPom = &parentPomItem
 					currentWrapper = jd.mavenWrapperPaths[i]
+					break
```

Comment:
How about adding unit test for this?

Reply:
Added UT and test data to cover the parent path detection for multi-level java projects.